src/pages/Register.js

```diff
@@ -0,0 +1,335 @@
+    }
+
+    this.handleStepZero = this.handleStepZero.bind(this)
+    this.handleStepOne = this.handleStepOne.bind(this)
+    this.handleVerification = this.handleVerification.bind(this)
+  }
+
+  next() {
+    const current = this.state.current + 1;
+    this.setState({ current });
+  }
+
+  prev() {
+    const current = this.state.current - 1;
+    this.setState({ current });
+  }
+
+  onLoginValueChange(key, val) {
+    this.setState({ [key]: val });
+  }
+
+  // step 0: email, password
+  handleStepZero(e) {
+    e.preventDefault();
+
+    this.props.form.validateFields(async (err, values) => {
+      if (!err) {
+        console.log('Received register values of form: ', values);
+
+        this.setState({
+          email: values['email'],
+          password: values['password']
+        })
+
+        this.next();
+      }
+    })
+  }
+
+  // step 1: create account
+  async handleStepOne(e) {
+    e.preventDefault();
+
+    this.setState({
+      registerLoading: true,
+      registerError: null
+    })
+
+    this.props.form.validateFields(async (err, values) => {
+      if (!err) {
+        console.log('Received register values of form: ', values);
+
+        const res = await api.register(
+          this.state.email,
+          this.state.password,
+          values['code']
+        )
+
+        if (res.code !== api.code.ok) {
+          let msg;
+          switch (res.code) {
+            case -1:
+              msg = "Wrong verification code"
+              break
+            case -2:
+              msg = "Wrong parameter"
+              break
+            case -6:
+              msg = "Account already exists"
+              break
+            default:
+              msg = "Unknown error";
+              break
+          }
+          this.setState({ registerError: msg })
+          message.info(msg)
+        } else {
+          this.next()
+        }
+        this.setState({ registerLoading: false })
+      }
+    })
+  }
+
+  // request verification code
+  async handleVerification (e) {
+    e.preventDefault();
+
+    this.setState({
+      verificationLoading: true,
+      verificationError: null
+    })
+
+    const res = await api.verification(this.state.email);
+
+    console.log("value: ", res)
+
+    if (res.code !== api.code.ok) {
+      let msg;
+      switch (res.code) {
+        case -2:
+          msg = "Wrong parameter";
+          break;
+        case -6:
+          msg = "Email already exists";
+          break;
+        default:
+          msg = "Unknown error";
+          break;
+      }
+      this.setState({ verificationError: msg })
+      message.info(msg);
+    } else {
+      message.info('Verification code has been sent!');
+    }
+    this.setState({ verificationLoading: false })
+  }
+
+  handleConfirmBlur = e => {
+    const { value } = e.target;
+    this.setState({ confirmDirty: this.state.confirmDirty || !!value });
+  }
+
+  compareToFirstPassword = (rule, value, callback) => {
```

Comment:
Also validate the length of the password?
Rule should be the same as https://github.com/websentry/WebSentry/pull/24